File: src/main/java/ninja/hudy/infosmog/model/CityRegion.java
package ninja.hudy.infosmog.model;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@DiscriminatorValue(RegionType.Values.CITY)
public class CityRegion extends Region {

    @OneToMany(mappedBy = "city")
    private Set<Station> stations = new HashSet<>();

    public Set<Station> getStations() {
        return stations;
    }

    public void setStations(Set<Station> stations) {
        this.stations = stations;
    }
}

File: src/main/kotlin/pl/org/smok/infosmog/model/CityRegion.kt
package pl.org.smok.infosmog.model

import pl.org.smok.infosmog.model.RegionType.Values.CITY
import javax.persistence.Entity
import javax.persistence.DiscriminatorValue
import javax.persistence.OneToMany
import java.util.HashSet

@Entity
@DiscriminatorValue(value = CITY)
class CityRegion : Region() {

    @OneToMany(mappedBy = "city")
    var stations: Set<Station> = HashSet()
}
